Document additions in Electron BaseApp (#444)
flatpak/flatpak-docs: docs/electron.rst

``org.electronjs.Electron2.BaseApp`` since ``branch/23.08`` comes with
``libnotify>=0.8.0``

.. _use-correct-desktop-filename:

Using correct desktop file name
-------------------------------

> It's important for Linux Applications to know the correct name of their desktop file. If not, it can led to problems e.g. a missing Icon under Wayland.

Reviewer: Nit to add empty lines after headings.

> It's important for Linux Applications to know the correct name of their desktop file. If not, it can led to problems e.g. a missing Icon under Wayland.

Reviewer 2: ```suggestion
It's important for Linux applications to know the correct name of their desktop file. If not, it can lead to problems e.g. a missing icon under Wayland.
```

Reviewer 2: There are some other unintended capitalizations below eg. App -> app, Icon -> icon

Reviewer 2: Also know is a bit weird, maybe the sentence should be: "It's important for Linux applications to set the correct desktop file name"

> It's important for Linux Applications to know the correct name of their desktop file. If not, it can led to problems e.g. a missing Icon under Wayland.

Reviewer 3: ```suggestion
It's important for Linux applications to know the correct name of their desktop file. If not, it can led to problems e.g. a missing icon under Wayland.
```

> By default electron uses ``{appname}.desktop`` as desktop file name. In Flatpak the name of the desktop file must be the id of the Flatpak.

Reviewer 2: ```suggestion
By default Electron uses ``{appname}.desktop`` as desktop file name. In Flatpak the name of the desktop file must be the id of the Flatpak.
```

Electron the framework is capitalized in the page

> To tell Electron to use another name you need to set ``desktopName`` key in your ``package.json`` e.g. ``"desktopName": "com.example.MyApp.desktop"``.

Reviewer 3: ```suggestion
To tell Electron to use another name you need to set the ``desktopName`` key in your ``package.json`` e.g. ``"desktopName": "com.example.MyApp.desktop"``.
```

In case you repack a binary, you can use the ``patch-desktop-filename`` script provided by the BaseApp. Each Electron binary ships with ``resources/app.asar`` file.
You need to call ``patch-desktop-filename`` with this file as Argument.

> If your App is e.g. installed under ``${FLATPAK_DEST}/my-app`` you need to run ``patch-desktop-filename ${FLATPAK_DEST}/my-app/resources/app.asar``.

Reviewer 3: ```suggestion
If your application is e.g. installed under ``${FLATPAK_DEST}/my-app`` you need to run ``patch-desktop-filename ${FLATPAK_DEST}/my-app/resources/app.asar``.
```

``org.electronjs.Electron2.BaseApp`` since ``branch/23.08`` comes with
``libnotify>=0.8.0``

.. _use-correct-desktop-filename:

Using correct desktop file name
-------------------------------

> It's important for Linux applications to set the correct desktop file name. If not, it can led to problems e.g. a missing icon under Wayland.

Reviewer 2: ```suggestion
It's important for Linux applications to set the correct desktop file name. If not, it can lead to problems e.g. a missing icon under Wayland.
```

.. code-block:: bash

  jq '.build.linux.target="dir"' <<<$(<package.json) > package.json

Make setProgressBar and setBadgeCount work
-------------------------------------------

> Electron has the 2 functions `setProgressBar <https://www.electronjs.org/de/docs/latest/api/browser-window#winsetprogressbarprogress-options>`_ and `setBadgeCount <https://www.electronjs.org/de/docs/latest/api/browser-window#winsetprogressbarprogress-options>`_.

Reviewer 2: ```suggestion
Electron has two functions `setProgressBar <https://www.electronjs.org/de/docs/latest/api/browser-window#winsetprogressbarprogress-options>`_ and `setBadgeCount <https://www.electronjs.org/de/docs/latest/api/browser-window#winsetprogressbarprogress-options>`_.
```

> These functions allow showing a Progress bar and a Number in the Window Icon. It is implemented under Linux using the UnityLauncherAPI. This API is not implemented on every Desktop. A known Desktop who implements this is KDE.

Reviewer 2: ```suggestion
These functions allow showing a Progress bar and a Number in the Window Icon. It is implemented under Linux using the UnityLauncherAPI. This API is not implemented on every Desktop. A known desktop environment which implements this is KDE.
```

It is also implemented by the popular `Dash to Dock <https://micheleg.github.io/dash-to-dock>`_ GNOME extension and `Plank <https://launchpad.net/plank>`_.

> To make it work in Flatpak, the App needs to :ref:`use the correct desktop filename <use-correct-desktop-filename>`. The Flatpak also need the ``--talk-name=com.canonical.Unity``.

Reviewer 2: ```suggestion
To make it work in Flatpak, the App needs to :ref:`use the correct desktop filename <use-correct-desktop-filename>`. The Flatpak also needs the ``--talk-name=com.canonical.Unity`` permission.
```

> Electron checks `checks if it's running on Unity or KDE <https://github.com/electron/electron/blob/fb88375ab4d2161dbf7e958a2a94c7c6d97dc84c/shell/browser/linux/unity_service.cc#L64>`_ before using the UnityLauncherAPI.

Reviewer: Linkcheck doesn't like this link... not sure how best to resolve that.

Author: Is there maybe a way to whitelist this?

Reviewer: I've not done it but maybe you can set `linkcheck_ignore = ['...']` in `conf.py`.

https://www.sphinx-doc.org/en/master/_modules/sphinx/builders/linkcheck.html

Reviewer 2: Better to use anchors_ignore I think https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-linkcheck_anchors_ignore

> To make this work on other Desktops too, you need to set ``XDG_CURRENT_SESSION=KDE`` to pretend the App is running on KDE.

Reviewer: Same comment about `ELECTRON_TRASH=gio`.

.. code-block:: bash

  jq '.build.linux.target="dir"' <<<$(<package.json) > package.json

Make setProgressBar and setBadgeCount work
-------------------------------------------
Electron has the two functions `setProgressBar <https://www.electronjs.org/de/docs/latest/api/browser-window#winsetprogressbarprogress-options>`_ and `setBadgeCount <https://www.electronjs.org/de/docs/latest/api/browser-window#winsetprogressbarprogress-options>`_.

> These functions allow showing a Progress bar and a Number in the Window Icon. It is implemented under Linux using the UnityLauncherAPI. This API is not implemented on every Desktop. A known desktop environment which implements this is KDE.

Reviewer 2: ```suggestion
These functions allow showing a progress bar and a number in the window icon. It is implemented under Linux using the UnityLauncherAPI. This API is not implemented on every Desktop. A known desktop environment which implements this is KDE.
```
